### PR TITLE
Update remaining actions

### PR DESCRIPTION
**Pull request by thejohnfreeman**

### CODE REVIEW

**Review by Bronek — APPROVED** [review]

File: .github/workflows/nix.yml [hunk 1]
```diff
@@ -83,7 +83,7 @@ jobs:
         with:
           configuration: ${{ matrix.configuration }}
       - name: upload archive
-        uses: actions/upload-artifact@v4
```
Comment (mvadari) [inline .github/workflows/nix.yml:86]: These look like they've been downgraded

File: .github/workflows/clang-format.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ jobs:
         git diff --exit-code | tee "clang-format.patch"
     - name: Upload patch
       if: failure() && steps.assert.outcome == 'failure'
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v3
```
Comment (mvadari) [inline .github/workflows/clang-format.yml:33]: Should this be v4?

File: .github/workflows/levelization.yml [hunk 3]
```diff
@@ -18,7 +18,7 @@ jobs:
         git diff --exit-code | tee "levelization.patch"
     - name: Upload patch
       if: failure() && steps.assert.outcome == 'failure'
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v3
```
Comment (mvadari) [inline .github/workflows/levelization.yml:21]: Should this be v4?

File: .github/workflows/nix.yml (hunk 1 above)
Comment (thejohnfreeman) [reply .github/workflows/nix.yml:86]: Deliberately so. The upgrade was just yesterday and some runners are failing to run v4. It is making the CI unreliable.

**Review by mvadari — APPROVED** [review]

File: .github/workflows/nix.yml (hunk 1 above)
Comment (thejohnfreeman) [reply .github/workflows/nix.yml:86]: [heavy-wsnnk-kcbnl](https://github.com/XRPLF/rippled/actions/runs/8269704525/job/22626252882)
[heavy-wsnnk-rmmwk](https://github.com/XRPLF/rippled/actions/runs/8269704525/job/22626252585)
[heavy-wsnnk-2dfzj](https://github.com/XRPLF/rippled/actions/runs/8266936245/job/22623669891)

Failing runners are running version 2.300.2. Passing runners are running 2.314.1. Until we get every runner on version >= 2.314.1, I think it's best to downgrade the action.

**Pull request merged**